dist/images/ovndb-raft-functions.sh

```diff
@@ -17,24 +17,59 @@ verify-ovsdb-raft() {
   fi
 }
 
-# OVN DB must be up in the first DB node
-# This waits for ovnkube-db-0 POD to come up
-ready_to_join_cluster() {
+bracketify() { case "$1" in *:*) echo "[$1]" ;; *) echo "$1" ;; esac }
+
+# checks if a db pod is part of a current cluster
+db_part_of_cluster() {
+  local pod=${1}
+  local db=${2}
+  local port=${3}
+  echo "Checking if ${pod} is part of cluster"
+  init_ip=$(kubectl --server=${K8S_APISERVER} --token=${k8s_token} --certificate-authority=${K8S_CACERT} \
+    get pod -n ${ovn_kubernetes_namespace} ${pod} -o=jsonpath='{.status.podIP}')
+  if [[ $? != 0 ]]; then
+    echo "Unable to get ${pod} ip "
+    return 1
+  fi
+  echo "Found ${pod} ip: $init_ip"
+  init_ip=$(bracketify $init_ip)
+  target=$(ovn-${db}ctl --timeout=5 --db=${transport}:${init_ip}:${port} ${ovndb_ctl_ssl_opts} --data=bare --no-headings \
+    --columns=target list connection)
+  if [[ "${target}" != "p${transport}:${port}${ovn_raft_conn_ip_url_suffix}" ]]; then
+    echo "Unable to check correct target ${target} "
+    return 1
+  fi
+
+  echo "${pod} is part of cluster"
+  return 0
+}
+
+# Checks if cluster has already been initialized.
+# If not it returns false and sets init_ip to ovnkube-db-0
+cluster_exists() {
   # See if ep is available ...
   local db=${1}
   local port=${2}
 
+  db_pods=$(kubectl --server=${K8S_APISERVER} --token=${k8s_token} --certificate-authority=${K8S_CACERT} \
+    get pod -n ${ovn_kubernetes_namespace} -o=jsonpath='{.items[*].metadata.name}' | egrep -o 'ovnkube-db[^ ]+')
+
+  for db_pod in $db_pods; do
+    if db_part_of_cluster $db_pod $db $port; then
+      echo "${db_pod} is part of current cluster with ip: ${init_ip}!"
+      return 0
```

Comment:
L60: this would mean that you could return out of the `for` loop for ovnkube-db-2 and that init_ip set ovkube-db-2 IP address